pkg/dataprotection/utils/utils.go

```diff
@@ -317,6 +318,20 @@ func GetPodFirstContainerPort(pod *corev1.Pod) int32 {
 	return ports[0].ContainerPort
 }
 
+func GetContainerPortByName(pod *corev1.Pod, containerName string, portName string) int32 {
+	for _, container := range pod.Spec.Containers {
+		if container.Name != containerName {
+			continue
+		}
+		for _, port := range container.Ports {
+			if port.Name == portName {
+				return port.ContainerPort
+			}
+		}
+	}
+	return 0
+}
```

Comment:
what if no such port is fouond?

Reply:
Should I check whether returned port number is valid(is not zero) or not, and skip adding env DP_DB_PORT if the value is zero?

Reply:
or if no such port is found, just return first containerPort.

Reply:
If the port + container specifed is invalid, we'd better raise an error.